clippy_lints/src/hash_borrow_str_soundness.rs

```diff
@@ -0,0 +1,82 @@
+use clippy_utils::diagnostics::span_lint_and_then;
+use clippy_utils::ty::implements_trait;
+use rustc_hir::def::{DefKind, Res};
+use rustc_hir::{Item, ItemKind, Path, TraitRef};
+use rustc_lint::{LateContext, LateLintPass};
+use rustc_middle::ty::Ty;
+use rustc_session::{declare_lint_pass, declare_tool_lint};
+use rustc_span::symbol::sym;
+
+declare_clippy_lint! {
+    ///
+    ///  This lint is concerned with the soundness of types that implement traits
+    ///  Hash, Borrow<u8> and Borrow<str>:
+    ///  ```
+    /// impl Hash for MyType {
```

Comment:
I would maybe move this example to its own `### Example` section.

Reply:
Maybe the example is a bit redundant?

Reply:
I'd keep it, It can hurt to have it imo. Though it's currently failing in CI. You can add `ignore` to the codeblock, since it looks like a visual example.

Reply:
I made the example complete to show that it breaks. I think that's better.